### PR TITLE
Mini challenge5/routing

### PR DESCRIPTION
**Pull request by dcastrowolf**
Here I leave the test coverage screenshot report. I'm still pending to improve testing across the project yet I hope this massive code illustrate the difficulty around time.

image

### CODE REVIEW

**Comment by dcastrowolf** [comment]
mock user credentials are:
username: wizeline
password: 123pormi

File: src/components/CardList/CardList.test.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import CardList from './CardList.component';
+
+const mockedVideoData = [
```
Comment (andresSaldanaAguilar) [inline src/components/CardList/CardList.test.jsx:6]: will recommend to extract this to mock files

File: src/components/Private/Private.component.jsx [hunk 2]
```diff
@@ -4,10 +4,15 @@ import { Route, Redirect } from 'react-router-dom';
...
+  } = useAuth();
 
   return (
-    <Route {...rest} render={() => (authenticated ? children : <Redirect to="/" />)} />
+    <Route
+      {...rest}
+      render={() => (authenticated ? children : <Redirect to="/login" />)}
+    />
   );
 }
```
Comment (andresSaldanaAguilar) [inline src/components/Private/Private.component.jsx:18]: pretty clever Daniel!

File: src/services/authenticationService.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  });
+};
+
+export const userStorage = credentialStorage();
+
+export const authenticationService = (() => {
+  const login = async ({ username, password }) => {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        if (username === 'wizeline' && password === '123pormi') {
```
Comment (andresSaldanaAguilar) [inline src/services/authenticationService.js:49]: this password made my day 😂

File: src/components/Card/Card.test.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { render, screen } from '@testing-library/react';
+import Card from './Card.component';
+
+const mockData = {
```
Comment (andresSaldanaAguilar) [inline src/components/Card/Card.test.jsx:6]: same here, extract this to a mock file to a cleaner test view

**Comment by andresSaldanaAguilar** [comment]
Great work Daniel 😄 , congratulations, thanks so much for your effort and such a good job! And yes, there is a lot of code implemented and it is hard to keep the percentage high, but this is the last part of this, you can achieve it for the final deliverable Daniel, I believe in you!
I just found an issue when a favorite video is played, the URL and request send undefined and the video player never resolves:
image

## Acceptance Criteria
- [x] All the sections have their own route.
- [x] Navigation across the sections is implemented correctly.
- [x] The Global State is persistent across all the sections.
- [x] The "Mocked Authentication" mechanism works correctly.
- [x] The "session data" is stored in the Global Context correctly.
- [x] Videos can be added to the Favorites list.
- [x] Videos can be removed from the Favorites list.
- [x] Navigation to **Favorite Videos View** using a private route is implemented correctly (only authenticated users should access this section).
- [] Navigation to the **Favorite Video Details View** using a private route is implemented correctly (only authenticated users should access this section). Couldn't reach the view
- [ ] Information for the selected favorite video is displayed correctly. Couldn't reach the view
- [x] The list of other favorite videos in the **Favorite Video Details View** is displayed.
- [ ] Testing coverage is above 70%. 

## Bonus Points
- [ ] The Add/remove from favorites button should appear when the user passes the mouse over the video card in the list.
- [ ] Integrate with a real authentication provider (such as Auth0, OAuth or Firebase).
- [x] A 404 section is shown when a route is not found.
- [ ] The **Login View** is implemented as a modal using React Portals.

**Pull request merged**